### PR TITLE
Adding Security Advisory 2021-001

### PR DESCRIPTION
**Pull request by ikedas**
This may fix sympa-community/sympa#1091 .

### CODE REVIEW

File: security/2021-001.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+There were the following problems with the use of this parameter.
+
+  1. This parameter, for its purpose, should be different for each installation,
+     and once set, it cannot be changed.
+     As a result, some sites have been operating with this parameter set to an
+     empty value.
+     This completely invalidates the security measures described above. 
+
+  2. even if the parameter is properly set, it may be considered not being
```
Comment (racke) [inline security/2021-001.md:47]: s/even/Even/

File: security/2021-001.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+     strong enough against brute force attacks.
+
+For the above reasons, administrators are recommended to do the following.
+
+Impact
```
Comment (racke) [inline security/2021-001.md:52]: Move this one up, otherwise "to do the following" is confusing.

File: security/2021-001.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+If you are operating without setting the `cookie` parameter and you cannot
+upgrade to the latest version of Sympa right now, set a value for that
+parameter.
+
+However, if you are using 6.2.40 or earlier, you need to upgrade your
+RC4-encrypted passwords by running `upgrade_sympa_password.pl` (with
+6.2.16 or later) or `sympa.pl --md5_encode_password` (earlier) before
+setting the parameter.
+
+
```
Comment (racke) [inline security/2021-001.md:70]: Restart sympa services (sympa, wwsympa, soap).

File: security/2021-001.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+Solution
+--------
+
+The root solution is to upgrade to Sympa 6.2.62 or later which no longer use
```
Comment (racke) [inline security/2021-001.md:74]: The best solution ... no longer uses ...

File: security/2021-001.md (hunk 2 above)
Comment (ikedas) [reply security/2021-001.md:52]: s/do the following/take the measures/

File: security/2021-001.md (hunk 4 above)
Comment (ikedas) [reply security/2021-001.md:74]: Thanks!

**Comment by ikedas** [comment]
"Workarounds" may mislead that setting the parameter may completely prevent attacks.  Should the section be changed to "Mitigation"?

**Comment by racke** [comment]
> "Workarounds" may mislead that setting the parameter may completely prevent attacks. Should the section be changed to "Mitigation"?

Yes, that sounds better :+1:

File: security/2021-001.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+Attacker can achieve XSS and so on in message archives.
+
+
+Workarounds
+-----------
+
+If you are operating without setting the `cookie` parameter and you cannot
+upgrade to the latest version of Sympa right now, set a value for this
+parameter to mitigate security risk.
```
Comment (racke) [inline security/2021-001.md:62]: Please use plural (security risks).

File: security/2021-001.md (hunk 2 above)
Comment (racke) [reply security/2021-001.md:51]: Maybe ... take the measures detailed below?

File: security/2021-001.md (hunk 5 above)
Comment (ikedas) [reply security/2021-001.md]: Done.

File: security/2021-001.md (hunk 2 above)
Comment (ikedas) [reply security/2021-001.md:52]: Done.

**Pull request merged**